File: src/my_project/easy_problems/from201to250/sum_digits_of_string_after_convert.py
from typing import List 

class Solution:
    def getLucky(self, s: str, k: int) -> int:
       
        
        def sum_digits(num: int) -> int:
            
            answer = 0
            quotient = 0
            
            num_to_string = str(num)
            len_num = len(num_to_string)
            
            for i in range(len_num):
                quotient = num // (10**(len_num -1 - i))
                answer += quotient
                num = num % (10**(len_num -1 - i))
                
            
            return answer 

                
        len_s = len(s)
        temp = ""       
        for i in range(len_s):
            temp += str(ord(s[i]) -96)
            
        print(temp)      
            
        answer = int(temp)
        
        print(answer)
            
        for l in range(k):
            answer = sum_digits(answer)
        
        
        return answer
    
    

solution = Solution()

# print(solution.getLucky(s = "iiii", k = 1))

# print(solution.getLucky( s = "leetcode", k = 2))

# print(solution.getLucky(s = "zbax", k = 2))

print(solution.getLucky("dbvmfhnttvr", 5))

File: src/my_project_typescript/easy_problems_typescript/from101to150/sumDigitsOfStringAfterConvert.ts
function getLucky(s: string, k: number): number {

    function sumDigits(num: number): number {
        let answer: number = 0
        let quotient: number = 0
        let num_to_string = num.toString()
        let lenNum: number = num_to_string.length

        for (let i = 0; i < lenNum; i++){
            quotient = Math.floor(num / (10**(lenNum -1 -i)))
            answer += quotient
            num = num % (10** (lenNum -1 -i))
        }

        return answer 
    }

    let lenS: number = s.length
    let temp: string = ""
    for (let i = 0; i < lenS; i++){
        temp += (s[i].charCodeAt(0) - 96).toString()
    }

    let answer: number = parseInt(temp)

    for (let l = 0; l < k; l++){
        answer = sumDigits(answer)
    }

    return answer
};


console.log(getLucky("iiii", 1))

console.log(getLucky("leetcode", 2))

console.log(getLucky("zbax", 2))


